crates/mongodb-connector/src/api_type_conversions/query_request.rs

```diff
@@ -381,89 +576,121 @@ fn v3_to_v2_expression(
             column,
             operator,
             value,
-        } => v3_to_v2_binary_comparison(scalar_types, column, operator, value),
-        v3::Expression::Exists {
-            in_collection,
-            predicate,
-        } => Ok(v2::Expression::Exists {
-            in_table: match in_collection {
-                v3::ExistsInCollection::Related {
-                    relationship,
-                    arguments: _,
-                } => v2::ExistsInTable::RelatedTable { relationship },
-                v3::ExistsInCollection::Unrelated {
-                    collection,
-                    arguments: _,
-                } => v2::ExistsInTable::UnrelatedTable {
-                    table: vec![collection],
+        } => v3_to_v2_binary_comparison(context, root_collection_object_type, object_type, column, operator, value),
+        v3::Expression::Exists { in_collection, predicate, } => {
+            let (in_table, collection_object_type) = match in_collection {
+                v3::ExistsInCollection::Related { relationship, arguments: _ } => {
+                    let v3_relationship = lookup_relationship(collection_relationships, &relationship)?;
+                    let v3_collection = context.find_collection(&v3_relationship.target_collection)?;
+                    let collection_object_type = context.find_object_type(&v3_collection.r#type)?;
+                    let in_table = v2::ExistsInTable::RelatedTable { relationship };
```

Comment:
Maybe need to capture a predicate here? But that is beyond the scope of propagating types so just something to think about following up on.

Reply:
I don't think so. In v3 there's no predicate on the `v3::ExistsInCollection::Related`, and in v2 additional predicates are already inlined into the overall predicate expression.

It's different in order by because order by isn't a predicate structure, so it needed a place to put additional non-join predicates (ie permissions predicates). Where clause predicates are a predicate structure so the permissions predicates are inlined into the overall predicate expression.